pkg/vm/serialization.go

```diff
@@ -0,0 +1,136 @@
+package vm
+
+import (
+	"errors"
+	"math/big"
+
+	"github.com/CityOfZion/neo-go/pkg/io"
+	"github.com/CityOfZion/neo-go/pkg/util"
+)
+
+type stackItemType byte
+
+const (
+	byteArrayT stackItemType = 0x00
+	booleanT   stackItemType = 0x01
+	integerT   stackItemType = 0x02
+	arrayT     stackItemType = 0x80
+	structT    stackItemType = 0x81
+	mapT       stackItemType = 0x82
+)
+
+func serializeItem(item StackItem) ([]byte, error) {
+	w := io.NewBufBinWriter()
+	serializeItemTo(item, w.BinWriter, make(map[StackItem]bool))
+	if w.Err != nil {
+		return nil, w.Err
+	}
+	return w.Bytes(), nil
+}
+
+func serializeItemTo(item StackItem, w *io.BinWriter, seen map[StackItem]bool) {
+	switch t := item.(type) {
+	case *ByteArrayItem:
+		w.WriteLE(byte(byteArrayT))
+		w.WriteBytes(t.value)
+	case *BoolItem:
+		w.WriteLE(byte(booleanT))
+		w.WriteLE(t.value)
+	case *BigIntegerItem:
+		w.WriteLE(byte(integerT))
+		w.WriteBytes(t.Bytes())
+	case *InteropItem:
+		w.Err = errors.New("not supported")
```

Comment:
It'd be nice to also check this branch in tests (and it will also make upper-level error path branches covered).